Assignment4/juanraromero-23A248/task08.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+"""Task08.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1BcckdbmJu3nwDRF3oEL6iAdOqlqhxBNA
+
+**Task 08: Completing missing data**
+"""
+
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+from rdflib import Graph, Namespace, Literal, URIRef
+g1 = Graph()
+g2 = Graph()
+g1.parse(github_storage+"/rdf/data01.rdf", format="xml")
+g2.parse(github_storage+"/rdf/data02.rdf", format="xml")
+
+"""Tarea: lista todos los elementos de la clase Person en el primer grafo (data01.rdf) y completa los campos (given name, family name y email) que puedan faltar con los datos del segundo grafo (data02.rdf). Puedes usar consultas SPARQL o iterar el grafo, o ambas cosas."""
+
+from rdflib.plugins.sparql import prepareQuery
+from rdflib.namespace import RDF, RDFS
+
+ns = Namespace("http://somewhere#")
+data = Namespace("http://data.org#")
+
+q1 = prepareQuery('''
+  SELECT ?individual ?property WHERE {
+     ?individual rdf:type data:Person.
+     ?individual ?property ?value.
+    }
+  ''',
+  initNs = { "rdfs": RDFS, "rdf":RDF, "data": data}
+)
+q2 = prepareQuery('''
+  SELECT ?individual ?property ?value WHERE {
+     ?individual rdf:type data:Person.
+     ?individual ?property ?value.
+    }
+  ''',
+  initNs = { "rdfs": RDFS, "rdf":RDF, "data": data}
+)
+print("All values")
+allValues= []
+values = []
+
+for r in g2.query(q2):
+  allValues.append((r.individual,r.property))
+  values.append((r.individual,r.property,r.value))
+
+print("Completed values")
+completed = []
+for r in g1.query(q1):
+  completed.append((r.individual,r.property))
+
+print("Comprobation")
+missing = []
+for (i,p) in allValues:
+  if (i,p) not in completed:
+    missing.append((i,p))
+
+for (i1,p1) in missing:
+  for (i2,p2,v) in values:
+    if i1 == i2 and p1 == p2:
+      g1.add((i1,p1,v))
+
```

Comment:
You are copying the KG in a variable, the operating with that variable. Why not use the KG directly?